Fix #851: Makes a tooltip appearing above the link

ygrishajev opened the pull request:
This is needed when user tries to create a link at the very bottom bound of the editor

jhchen commented on ui/tooltip.js:
```diff
@@ -38,6 +38,10 @@ class Tooltip {
       this.root.style.left = (left + shift) + 'px';
     }
+    if (rootBounds.bottom > containerBounds.bottom) {
+      shift = containerBounds.bottom - rootBounds.bottom - this.root.offsetHeight;
```
`shift` is already being used for horizontal shifting and returned

ygrishajev commented:
@jhchen Ok, I changed `shift` into `horizontalShift` to better reflect its sence and added `horizontalShift` to represent the relevant action. Also changed `position()` method to return an object containing vertical and horisontal shifts.

jhchen commented:
Thank you for the updates. To be clear it was not a stylistic preference in the naming. The name was already taken and overwriting it creates bugs. Similarly changing the function signature of `position()` also creates bugs in the bubble theme which rely on the `position()` function. Can you just have `position()` return the horizontal shift as an integer?

ygrishajev commented:
@jhchen Thanks for the explanation. Didn't notice `position()` use in bubble theme. Made the updates needed.

jhchen commented:
Thank you for the changes. I did some testing and since the Bubble theme has an arrow, it should be flipped when the tooltip is shifted above the text. Also during scrolling, ideally the tooltip would know to flip as room above and below the highlighted text changes within the viewport. It's not obvious to me how these issues can be resolved so I'll have to take another look when I have more time. But feel free to dig in and make suggestions. If you set up the [development environment](https://github.com/quilljs/quill/blob/develop/.github/DEVELOPMENT.md) you can test a standalone bubble page with localhost:9000/standalone/bubble

cspanring commented:
We just ran into this issue with the (default) snow theme: our users can't edit links on lines close to the bottom of the editor.

Would it be possible to merge this PR as it fixes the issue with the default snow theme and open a separate issue for the Bubble theme?

jhchen commented:
No the issues I listed has not been resolved. I do not have an implementation recommendation for fixing them but they do need to be fixed before accepting this PR.

jhchen commented:
This has been fixed in 1.1.9
